larrecodnn/ImagePatternAlgs/Tensorflow/Modules/RawWaveformDump_module.cc

```diff
@@ -472,6 +475,13 @@ nnet::RawWaveformDump::analyze(art::Event const& evt)
           rndm_dist(rndm_engine); // randomly select one channel with a signal from this particle
         chnum = ittrk.second[i];
 
+        if (std::find(SelectedChannel.begin(), SelectedChannel.end(), chnum) != SelectedChannel.end()) {
+          continue; // channel already used
+        }
+        else {
+          SelectedChannel.push_back(chnum);
+        }
+
```

Comment:
Using the `selected_channels` object, the above lines can be replaced with:
```.cpp
if (not selected_channels.insert(chnum).second) {
  continue;
}
```
The above code will attempt to insert the `chnum` value into `selected_channels`.  If the insertion succeeds, then the value of `.second` will be `true`.  If the insertion fails, then `.second` will return `false`.

Reply:
Hi Kyle, thank you very much for your comments. I already replaced the codes as you recommended. I did learn a lot from your suggestions.